panther_docking/src/dock_pose_publisher_node.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2024 Husarion Sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+#include <rclcpp/rclcpp.hpp>
+
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+
+class DockPosePublisherNode : public rclcpp::Node
+{
+public:
+  DockPosePublisherNode() : Node("pose_publisher_node")
+  {
+    declare_parameter("publish_rate", 10.0);
+    declare_parameter("docks", std::vector<std::string>({"main"}));
+    declare_parameter("fixed_frame", "odom");
+
+    const auto fixed_frame = get_parameter("fixed_frame").as_string();
+    const auto docks = get_parameter("docks").as_string_array();
+
+    std::string dock_pose_frame_id;
+    for (const auto & dock : docks) {
+      declare_parameter(dock + ".type", "panther_charging_dock");
+      declare_parameter(dock + ".frame", "main_wibotic_receiver_requested_pose_link");
+
+      const auto dock_type = get_parameter(dock + ".type").as_string();
+      if (dock_type == "panther_charging_dock") {
+        dock_pose_frame_id = get_parameter(dock + ".frame").as_string();
+      }
+    }
+
+    const auto publish_rate = get_parameter("publish_rate").as_double();
+    const auto publish_period = std::chrono::duration<double>(1.0 / publish_rate);
+
+    source_frame_ = dock_pose_frame_id;
+    target_frame_ = fixed_frame;
+
+    pose_publisher_ = this->create_publisher<geometry_msgs::msg::PoseStamped>(
+      "docking/dock_pose", 10);
+
+    tf_buffer_ = std::make_unique<tf2_ros::Buffer>(this->get_clock());
+    tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+
+    timer_ = this->create_wall_timer(
+      publish_period, std::bind(&DockPosePublisherNode::publishPose, this));
+  }
+
+private:
+  void publishPose()
+  {
+    geometry_msgs::msg::TransformStamped transform_stamped;
+    try {
+      transform_stamped = tf_buffer_->lookupTransform(
```

Comment:
Imu source_frame should be webotic receiver frame. Thanks that we can mount it e.g. on the rear and it should also works. Also it si more intuitive to specified distance between receiver and station than station and base_link.
Also right now, probably this code doesn't work if we mount receiver on lynx, because the distance between station and base_link will be too large.

Reply:
We discussed this on priv. We will make it in the next version.